qln/init.go

```diff
@@ -136,6 +136,23 @@ func NewLitNode(privKey *[32]byte, path string, trackerURL string, proxyURL stri
 	nd.PeerMap = map[*lnp2p.Peer]*RemotePeer{}
 	nd.PeerMapMtx = &sync.Mutex{}
 
+	// Do we do autoreconnection things?
+	if autoreconn {
```

Comment:
Two things need to change here:

1. Don't block the main thread here. Should do this in a goroutine in stead.
2. Using the cached endpoint seems pointless. People can move around, or have a dynamic IP. Plus, the outgoing port will never match the port that lit actually listens on. So the cached endpoint is mostly useless due to NAT. Use the tracker for lookup, if someone is not in the tracker, fail.

Reply:
https://github.com/gertjaap/lit/commit/5380bfd007711c792ef7dc26112d6ebc639766fa

Reply:
Really it should attempt to connect with the known IP address, and if it fails it should do a lookup and try again with the new IP address.  And that should be part of lnp2p, but that's a project for another day.

Reply:
Fixed.